src/Validator/Scope/ScopeVisibilityValidator.php

```diff
@@ -77,7 +77,9 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
-        if (!$this->managedZoneProvider->isManagerOfZone($this->getAdherent(), $scopeGenerator->getCode(), $value->getZone())) {
+        if (!$this->managedZoneProvider->zoneBelongsToSome($value->getZone(), array_map(
```

Comment:
What about creating `zoneBelongsToSomeZomes($value->getZone(), $scope->getZones())`
and doing `array_map` on $zones inside? We will avoid duplication of `static function (Zone $zone) { return $zone->getId(); }, $scope->getZones())`